Tests/Shared/Core/UnitTestBase.swift

```diff
@@ -108,29 +108,42 @@ open class UnitTestBase<Component>: XCTestCase {
         diGraphShared.reset()
     }
 
+    // All data that the SDK writes, delete it here so each test function has a clean environment to run and does not depend on the result of the previous test.
     open func deleteAllPersistentData() {
-        var expectations: [XCTestExpectation] = []
-
-        let resetEventBusExpectation = XCTestExpectation(description: "reset EventBus to initial state")
-        expectations.append(resetEventBusExpectation)
-        Task {
-            await diGraphShared.eventBusHandler.reset()
```

Comment:
I do not believe we need to because all objects in the digraph are re-created before each test function. This includes the eventbus so all observers would be removed when we re-create the eventbus.